lighthouse-core/gather/gatherers/source-maps.js

```diff
@@ -0,0 +1,155 @@
+    /** @type {string} */
+    const sourceMapJson =
+      await driver.evaluateAsync(`(${fetchSourceMap})(${JSON.stringify(sourceMapUrl)})`);
+    return JSON.parse(sourceMapJson);
+  }
+
+  /**
+   * @param {string} sourceMapURL
+   * @return {LH.Artifacts.RawSourceMap}
+   */
+  parseSourceMapFromDataUrl(sourceMapURL) {
+    const buffer = Buffer.from(sourceMapURL.split(',')[1], 'base64');
+    return JSON.parse(buffer.toString());
+  }
+
+  /**
+   * @param {LH.Crdp.Debugger.ScriptParsedEvent} event
+   */
+  onScriptParsed(event) {
+    if (event.sourceMapURL) {
+      this._scriptParsedEvents.push(event);
+    }
+  }
+
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   */
+  async beforePass(passContext) {
+    const driver = passContext.driver;
+    driver.on('Debugger.scriptParsed', this.onScriptParsed);
+    await driver.sendCommand('Debugger.enable');
+  }
+
+  /**
+   * @param {string} url
+   * @param {string} base
+   */
+  _resolveUrl(url, base) {
+    try {
+      return new URL(url, base).href;
+    } catch (e) {
+      return url;
+    }
+  }
+
+  /**
+   * @param {Driver} driver
+   * @param {LH.Crdp.Debugger.ScriptParsedEvent} event
+   * @return {Promise<LH.Artifacts.SourceMap | undefined>}
+   */
+  async _processEvent(driver, event) {
+    if (!event.sourceMapURL) {
+      return;
+    }
+
+    // `sourceMapURL` is simply the URL found in either a magic comment or an x-sourcemap header.
+    // It has not been resolved to a base url.
+    const isSourceMapADataUri = event.sourceMapURL.startsWith('data:');
+    const sourceMapUrl = isSourceMapADataUri ?
+        event.sourceMapURL :
+        this._resolveUrl(event.sourceMapURL, event.url);
+
+    log.verbose('SourceMaps', event.url, URL.elideDataURI(sourceMapUrl));
+    /** @type {{map: LH.Artifacts.RawSourceMap} | {errorMessage: string}} */
+    let sourceMapOrError;
+    try {
+      const map = isSourceMapADataUri ?
+        this.parseSourceMapFromDataUrl(sourceMapUrl) :
+        await this.fetchSourceMapInPage(driver, sourceMapUrl);
+      sourceMapOrError = {map};
+    } catch (err) {
+      sourceMapOrError = {errorMessage: err.toString()};
+    }
+
+    if ('errorMessage' in sourceMapOrError) {
+      log.error('SourceMaps', event.url, sourceMapOrError.errorMessage);
+    }
+
+    return {
+      scriptUrl: event.url,
+      sourceMapUrl: isSourceMapADataUri ? undefined : sourceMapUrl,
+      ...sourceMapOrError,
+    };
+  }
+
+  /**
+   * @param {LH.Gatherer.PassContext} passContext
+   * @return {Promise<LH.Artifacts['SourceMaps']>}
+   */
+  async afterPass(passContext) {
+    const driver = passContext.driver;
+
+    driver.off('Debugger.scriptParsed', this.onScriptParsed);
+    await driver.sendCommand('Debugger.disable');
+
+    const eventProcessPromises = this._scriptParsedEvents
+      .filter((event) => event.sourceMapURL)
+      .map((event) => this._processEvent(driver, event));
+
+    return [...await Promise.all(eventProcessPromises)]
```

Comment:
`(await Promise.all(eventProcessPromises))` is already an array so doesn't need the spread, but needs the parens